tasks/main.yml

```diff
@@ -45,6 +45,11 @@
   apt: name=oracle-java8-unlimited-jce-policy
   when: install_cryptography_extension|bool
 
+- name: Set Java keystore password
+  shell: keytool -storepasswd -new {{ ca_certificates.password }} -storepass changeit -keystore /usr/lib/jvm/java-8-oracle/jre/lib/security/cacerts && touch /root/.keystore-pass-set
```

Comment:
Should not `{{ ca_certificates.password }}` reused there ?

Reply:
The `-storepass` value is the old value (ie. the default `changeit`). I already use `{{ ca_certificates.password }}` for the new value, so it seems ok to me.

Reply:
There is a way to avoid to create a canary file: we could try to open the keystore using the default password and if it succeeded change the password:
```
- name: check if default password is used
  command: keytool -list -keystore /usr/lib/jvm/java-8-oracle/jre/lib/security/cacerts -storepass changeit
  register: keytool_list

- name: change keystore password
  command:
    keytool -storepasswd -new {{ ca_certificates.password }} -storepass changeit -keystore /usr/lib/jvm/java-8-oracle/jre/lib/security/cacerts
  when: keytool_list is success
``` 

Side notes: a default role variable (might be named `default_keystore_password`) could be used instead of `changeit`.

Reply:
Ok, makes more sense to me ! I'm going to implement your solution, it's better if we can avoid using a canary file.